main.go

```diff
@@ -280,6 +284,9 @@ func loadConfig() (*config.Config, error) {
 }
 
 func applyConfig(cfg *config.Config) error {
+	if proxy == nil {
+		proxy = newReverseProxy(cfg.Server.Proxy.SkipTLSVerify)
```

Comment:
since the Proxy part is optional in the configuration, this line looks suspicious.

Reply:
Okay, I took another pass at this, but I'm open to feedback here: It's not entirely clear to me how dynamic you _want_ this behavior to be?  The current tests in main_test don't reset server state between them so I tried to make the behavior work unsurprisingly with that.

Reply:
my bad I wasn't explicit enough. Since is Proxy is optional in the conf, my guess is that cfg.Server.Proxy can be nil so using cfg.Server.Proxy.SkipTLSVerify could lead to some crashes. So it better to check if cfg.Server.Proxy is not nil before using cfg.Server.Proxy.SkipTLSVerify. I'm not 100% sure but it's worth digging

Reply:
I _think_ that the current check for this using reflect.ValueOf().IsZero is correct: as long as you initialize the config struct the inner structs should be there (non-nil) with their zero values -- if that weren't the case I'd expect `TestReloadConfig` to panic on L291 or L298 since `testdata/http.yml` doesn't have a proxy section.

Reply:
I've re-arranged the logic in applyConfig to make it clearer what's going on.

Reply:
my bad,
I just checked the code of config.go and even if the Proxy is optional in the yaml, it's always present in the struct since it has this form
`type Server struct {
[...]
	// Optional Proxy configuration
	Proxy Proxy `yaml:"proxy,omitempty"`
[...]
}`
I don't know why but I thought the struct had this form (and since it was the weekend I just quickly looked at your PR)
`type Server struct {
[...]
	// Optional Proxy configuration
	Proxy *Proxy `yaml:"proxy,omitempty"`
[...]
}`
So you can rollback to the first version of the applyConfig since it's easier to read and safe (despite what I said in my first comment, sorry about that).

Reply:
No worries -- done!